packages/protocol/contracts/common/GoldToken.sol

```diff
@@ -91,6 +94,17 @@ contract GoldToken is
     return succeeded;
   }
 
+  /**
+   * @notice This function allows a user to burn a specific amount of tokens.
+     Burning is implemented by sending tokens to the burn address.
+   * @param value: The amount of CELO to burn.
+   * @return True if burn was successful.
+   */
+  function burn(uint256 value) external returns (bool) {
+    // not using transferWithCheck as the burn address can potentially be the zero address
+    return _transfer(BURN_ADDRESS, value);
```

Comment:
Nit: seems like it might be cleaner to call this `_transferWithoutCheck `  and keep the behavior of `_transfer` unchanged